kernel/src/log_segment.rs

```diff
@@ -446,11 +454,19 @@ fn list_log_files(
         }))
 }
 
+/// A struct to hold the result of listing log files. The commit and compaction files are guaranteed
+/// to be sorted in ascending order by version. The elements of `checkpoint_parts` are all the parts
+/// of the same checkpoint. Checkpoint parts share the same version.
+// TODO: debug asserts in a `new` method to enforce the above?
```

Comment:
i like this :) add issue if we want this to be taken up as a good-first-issue?

Reply:
Yeah, I can create one when this merges